base/array.jl

```diff
@@ -1169,24 +1171,71 @@ and [`prepend!`](@ref) and [`pushfirst!`](@ref) for the opposite order.
 """
 function append! end
 
-function append!(a::Vector, items::AbstractVector)
+append!(a::AbstractVector, iter) = _append!(a, IteratorSize(iter), iter)
+append!(a::AbstractVector{T}, it1::T, it2::T, iter::Vararg{T}) where T = foldl(_append_converted!, (it1, it2, iter...); init=a)
```

Comment:
This is intended for things like `append!(Int[], 1, 2)` (which was legal in the original code), but is technically bad/wrong, because `it1` and `it2` have the eltype of the container, and are not actually containers themselves. I think this could/should be removed as a bug, but that likely needs a PkgEval.

Reply:
What isn't this like `append!(a::AbstractVector{T}, it::T, iter::Vararg{T}) where T`?

Reply:
Good question - that's likely a leftover from before I added the tuple method. I seem to remember there was also some weird test failure. I'll change it and see if the tests run through!